src/store.rs

```diff
@@ -72,6 +73,11 @@ impl TicketStore {
     pub fn update_ticket_status(&mut self, id: TicketId, status: Status) -> Option<()> {
         self.data.get_mut(&id).map(|t| t.status = status)
     }
+
+    pub fn add_comment_to_ticket(&mut self, id: TicketId, comment: String) -> Option<()> {
```

Comment:
This function will panic if you pass in an empty string as a comment. 
I'd suggest to push validation at the edges: require `comment` to be of type `Comment`, hence you can rely on the invariant that it's going to be non-empty and you don't have to perform any validation.

Reply:
I think I'm on the same page. I've updated add_comment_to_ticket to take a Comment instead of string.